### PR TITLE
Figure.plot3d: Deprecate parameter "sizes" to "size" (remove in v0.6.0)

### LINKED ISSUE #1118: Rename the sizes parameter in plot/plot3d to be size
**Description of the problem**

Both [plot](https://www.pygmt.org/latest/api/generated/pygmt.Figure.plot.html#pygmt.Figure.plot) and [plot3d](https://www.pygmt.org/latest/api/generated/pygmt.Figure.plot3d.html#pygmt.Figure.plot3d) have the `sizes` parameter that are used to plot symbols with different sizes. However, `size` is used in GMT documentation. Shall we rename this parameter to be `size`? If yes, I think we should do it as early as possible cause this parameter is often used in PyGMT.

Originally posted in https://github.com/GenericMappingTools/pygmt/issues/1076#issuecomment-804986728: 
> > like the `sizes` parameter (I dont' understand why we add a **s** at the end) of the `plot`/`plot3d` method
> 
> `plot` is one of the oldest wrappers in PyGMT, and the parameters are not carefully chosen. We had some discussions in the GMT Community meeting, and I think we agree that we will change `sizes` to `size`.

----

See the format of the [**-S**](https://docs.generic-mapping-tools.org/latest/plot.html#s) option in GMT `plot`:

> x y [ z ] [ size ] [ symbol-parameters ] [ symbol ]

See the format of the [**-S**](https://docs.generic-mapping-tools.org/latest/plot3d.html#s) option in GMT `plot3d`:

> x y z [ w ] [ size ] [ symbol-parameters ] [ symbol ]

GMT plot | PyGMT plot | GMT plot3d | PyGMT plot3d
:---: | :---:|:---: | :---:
*z* | `color` | *w* | `color`
*size* | `sizes` | *size* | `sizes`
*intensity* | `intensity` |*intensity* | `intensity` 
*symbol* | `symbol` (requested in #1076) | *symbol* | `symbol` (requested in #1076)

---

Now, `sizes` is used in the following files:
- `src/plot.py`
- `tests/test_plot.py`
- `examples/tutorials/plot.py`
- `examples/gallery/symbols/scatter.py`
- `examples/gallery/symbols/points_categorical.py`
- `src/plot3d.py`
- `tests/test_plot3d.py`
- `examples/gallery/3d_plots/scatter3d.py`

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Closes #1118.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/plot3d.py [hunk 1]
```diff
@@ -179,12 +181,12 @@ def plot3d(
             )
         extra_arrays.append(kwargs["G"])
         del kwargs["G"]
-    if sizes is not None:
+    if size is not None:
         if kind != "vectors":
             raise GMTInvalidInput(
-                "Can't use arrays for sizes if data is matrix or file."
+                "Can't use arrays for 'size' if data is a matrix or file."
```
Comment (michaelgrund) [inline pygmt/src/plot3d.py:187]:
```suggestion
                "Can't use arrays for 'size' if data is a matrix or a file."
```
Feel free to change it but I'm also fine if you leave it as is.

**Review by core-man — APPROVED** [review]
Looks good~

File: pygmt/tests/test_plot3d.py [hunk 2]
```diff
@@ -458,3 +459,31 @@ def test_plot3d_scalar_xyz():
...
+@pytest.mark.mpl_image_compare(filename="test_plot3d_sizes.png")
+def test_plot3d_deprecate_sizes_to_size(data, region):
+    """
+    Make sure that the old parameter "sizes" is supported and it reports an
+    warning.
+
+    Modified from the test_plot3d_sizes() test.
+    """
+    fig = Figure()
+    with warnings.catch_warnings(record=True) as w:
```
Comment (weiji14) [inline pygmt/tests/test_plot3d.py:473]:
```suggestion
    with pytest.warns(expected_warning=FutureWarning) as record:
```

File: pygmt/tests/test_plot3d.py [hunk 3]
```diff
@@ -458,3 +459,31 @@ def test_plot3d_scalar_xyz():
...
+            perspective=[225, 30],
+            sizes=0.5 * data[:, 2],
+            region=region,
+            projection="X10c",
+            style="ui",
+            color="blue",
+            frame=["af", "zaf"],
+        )
+        assert len(w) == 1
+        assert issubclass(w[0].category, FutureWarning)
```
Comment (weiji14) [inline pygmt/tests/test_plot3d.py:488]:
```suggestion
        assert len(record) == 1  # check that only one warning was raised
```

File: pygmt/tests/test_plot3d.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 Tests plot3d.
 """
 import os
+import warnings
```
Comment (weiji14) [inline pygmt/tests/test_plot3d.py:5]:
```suggestion
```

**Review by weiji14 — COMMENTED** [review]
Just a similar suggestion to my comment on https://github.com/GenericMappingTools/pygmt/pull/1254#discussion_r627769349 to use `pytest.warns` (see also https://docs.pytest.org/en/6.2.x/warnings.html#warns).

**Review by weiji14 — APPROVED** [review]

**Pull request merged**